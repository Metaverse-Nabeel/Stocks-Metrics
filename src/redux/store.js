/* eslint-disable import/no-extraneous-dependencies */
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import cryptoReducer from './cryptoSlice/cryptoSlice';
import detailsReducer from './cryptoSlice/cryptoDetailsSlice';

const reducer = combineReducers({
  crypto: cryptoReducer,
  details: detailsReducer,
});

const store = configureStore({ reducer });

export default store;
