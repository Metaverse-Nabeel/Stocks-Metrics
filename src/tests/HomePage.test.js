import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import HomePage from '../routes/homePage';

describe('Test for the home page', () => {
  test('Home Page rendering correct', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <HomePage />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
