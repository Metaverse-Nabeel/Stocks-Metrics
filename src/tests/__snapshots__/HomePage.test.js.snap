// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the home page Home Page rendering correct 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <input
          placeholder="Type here..."
          type="text"
          value=""
        />
      </div>
      <h3>
        Top 50 Crypto Currencies
      </h3>
      <div>
        <div>
          <h3>
            Crypto Not Found
          </h3>
          <img
            alt="Image404Error"
            src=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <input
        placeholder="Type here..."
        type="text"
        value=""
      />
    </div>
    <h3>
      Top 50 Crypto Currencies
    </h3>
    <div>
      <div>
        <h3>
          Crypto Not Found
        </h3>
        <img
          alt="Image404Error"
          src=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
